entities-viewings-collector/src/main/scala/io/renku/entities/viewings/collector/projects/EventPersister.scala

```diff
@@ -51,7 +51,15 @@ private[viewings] class EventPersisterImpl[F[_]: MonadThrow](tsClient: TSClient[
   override def persist(event: ProjectViewedEvent): F[Unit] =
     findProjectId(event) >>= {
       case None            => ().pure[F]
-      case Some(projectId) => deleteOldViewedDate(projectId) >> insert(projectId, event)
+      case Some(projectId) => persistIfOlderOrNone(event, projectId)
+    }
+
+  private def persistIfOlderOrNone(event: ProjectViewedEvent, projectId: projects.ResourceId) =
+    findStoredDate(projectId) >>= {
+      case None => insert(projectId, event)
+      case Some(date) if (date compareTo event.dateViewed) < 0 =>
```

Comment:
I think using infix is supported for comparables, like `date < event.dateViewed` (not important ofc, just maybe reads a bit easier :))

Reply:
👍🏻 Changed